addons/pos_coupon/static/src/xml/OrderWidget.xml

```diff
@@ -3,7 +3,33 @@
 
     <t t-name="pos_coupon.OrderWidget" t-inherit="point_of_sale.OrderWidget" t-inherit-mode="extension" owl="1">
         <xpath expr="//OrderSummary" position="after">
-            <ActivePrograms t-props="getActiveProgramsProps()" />
+            <div class="active-programs">
+                <t t-set="_props" t-value="this.getActiveProgramsProps()"/>
```

Comment:
I don't like the `_props` naming because it's confusing and it's not properties passed from outside.
What I can suggest is to either use for each fields a separate getter or compute that in the `onRender` hook. You can ask JCB for further info for that last advice.

Reply:
Hi @trj-odoo, thanks for your feedback. :) I opted for the `onWillRender` hook solution that you suggested. Are my changes similar to what you had in mind? Have a look and let me know what you think! Thank you!